paymentsheet/src/test/java/com/stripe/android/customersheet/utils/CustomerSheetHacksTest.kt

```diff
@@ -27,8 +29,26 @@ class CustomerSheetHacksTest {
     }
 
     @Test
-    fun `on initialize, should initialize all data sources as expected`() = runTest {
-        CustomerSheetHacks.initialize(application, TestLifecycleOwner(), FakeCustomerAdapter())
+    fun `on initialize with adapter, should initialize all data sources as expected`() = runTest {
+        CustomerSheetHacks.initialize(
+            application = application,
+            lifecycleOwner = TestLifecycleOwner(),
+            integrationType = CustomerSheetIntegrationType.Adapter(FakeCustomerAdapter())
+        )
+
+        assertThat(CustomerSheetHacks.initializationDataSource.awaitWithTimeout()).isNotNull()
+        assertThat(CustomerSheetHacks.intentDataSource.awaitWithTimeout()).isNotNull()
+        assertThat(CustomerSheetHacks.paymentMethodDataSource.awaitWithTimeout()).isNotNull()
+        assertThat(CustomerSheetHacks.savedSelectionDataSource.awaitWithTimeout()).isNotNull()
```

Comment:
Thinking we should validate the instance injected here (ie for adapter they should all be `CustomerAdapterDataSource`)

Reply:
sgtm! You could probably do this with assertThat(<datasource>).isInstanceOf<CustomerAdapterDataSource>, right?